src/backend/catalog/namespace.c

```diff
@@ -1982,6 +1982,13 @@ recomputeNamespacePath(void)
 			elog(DEBUG3, "recompute search_path[%s] when acl_type is ranger", namespace_search_path);
 		}
 	}
+	else 
+	{
+		if (aclType == HAWQ_ACL_RANGER && debug_query_string != NULL)
+		{
+			last_query_sign = string_hash(debug_query_string, strlen(debug_query_string));
+		}
+	}
```

Comment:
Thx!
If there are other code should be changed also, I will try to clear the logic.